test/unit/query_expr_compiler/test_rewrite_rules.py

```diff
@@ -0,0 +1,390 @@
+    catalog: Catalog,
+    query_mechanism: str,
+    output_measure: Union[PureDP, ApproxDP, RhoZCDP],
+    expected_mechanism: str,
+) -> None:
+    """Test noise selection for GroupByCount query expressions."""
+    expr = SuppressAggregates(
+        child=GroupByCount(
+            child=BASE_EXPR,
+            groupby_keys=KeySet.from_dict({}),
+            mechanism=CountMechanism[query_mechanism],
+        ),
+        column="count",
+        threshold=42,
+    )
+    info = CompilationInfo(output_measure=output_measure, catalog=catalog)
+    got_expr = select_noise_mechanism(info)(expr)
+    assert got_expr == replace(
+        expr,
+        child=replace(expr.child, core_mechanism=NoiseMechanism[expected_mechanism]),
+    )
+
+
+@parametrize(
+    [
+        Case()(
+            expr=GroupByCount(
+                child=BASE_EXPR,
+                groupby_keys=KeySet.from_dict({}),
+                mechanism=CountMechanism.GAUSSIAN,
+            )
+        ),
+        Case()(
+            expr=GroupByCountDistinct(
+                child=BASE_EXPR,
+                groupby_keys=KeySet.from_dict({}),
+                mechanism=CountDistinctMechanism.GAUSSIAN,
+            )
+        ),
+        Case()(
+            expr=GroupByBoundedAverage(
+                child=BASE_EXPR,
+                groupby_keys=KeySet.from_dict({}),
+                measure_column="int_col",
+                low=0,
+                high=1,
+                mechanism=AverageMechanism.GAUSSIAN,
+            )
+        ),
+        Case()(
+            expr=GroupByBoundedSum(
+                child=BASE_EXPR,
+                groupby_keys=KeySet.from_dict({}),
+                measure_column="int_col",
+                low=0,
+                high=1,
+                mechanism=SumMechanism.GAUSSIAN,
+            )
+        ),
+        Case()(
+            expr=GroupByBoundedSTDEV(
+                child=BASE_EXPR,
+                groupby_keys=KeySet.from_dict({}),
+                measure_column="int_col",
+                low=0,
+                high=1,
+                mechanism=StdevMechanism.GAUSSIAN,
+            )
+        ),
+        Case()(
+            expr=GroupByBoundedVariance(
+                child=BASE_EXPR,
+                groupby_keys=KeySet.from_dict({}),
+                measure_column="int_col",
+                low=0,
+                high=1,
+                mechanism=VarianceMechanism.GAUSSIAN,
+            )
+        ),
+        Case()(
+            expr=SuppressAggregates(
+                child=GroupByCount(
+                    child=PrivateSource("blah"),
+                    groupby_keys=KeySet.from_dict({}),
+                    mechanism=CountMechanism.GAUSSIAN,
+                ),
+                column="count",
+                threshold=42,
+            )
+        ),
+    ],
+)
+def test_noise_selection_invalid_noise(catalog: Catalog, expr: QueryExpr) -> None:
+    for output_measure in (PureDP(), ApproxDP()):
```

Comment:
nit: Can we put these into the parameters? I was reading through them wondering "why are these invalid"?